openassessment/management/commands/create_oa_submissions_from_file.py

```diff
@@ -190,18 +193,22 @@ def get_usernames(self, submissions_config):
 
         return learners, course_staff
 
-    def load_anonymous_user_ids(self, course_id, usernames):
-        """
-        Look up all Users with the given usernames, look up their anonymous user ids for the specified course, and
-        store a mapping from username to anonaymous id in self.username_to_anonymous_user_id
-        """
+    def _import_anonymous_id_for_user(self):
```

Comment:
Interesting... I don't really agree with this refactor.

**Reasoning** - I sort of feel like it's *more* bizarre to mock private internals of a class than mocking the imports. The imports are *supposed* to be these separate, modular, things which can be mocked because they exist outside your class under test, whereas mocking an internal function of the class under test inherently ties your tests to the internal structure of the class which makes them brittle and leaks implementation into your test.

Reply:
Generally yes - if this was a top level import, and I could do the "mock it where it's used" mock()-ing as intended and just do `mock(...create_oa_submissions_from_file.anonymous_id_for_user)` then yes, I agree.

However, since these are inline imports that won't be able to resolve when running tests isolated from platform, when I say "mocking imports" I don't mean mocking the imported value (sorry, definitely not clear) I mean literally mocking python's import logic. doing a patch over the python import internals. THAT's the hammer.